app/assets/javascripts/app/views/comment_stream_view.js

```diff
@@ -62,14 +61,41 @@ app.views.CommentStream = app.views.Base.extend({
     }
   },
 
+  insertPoint: 0, // An index of the comment added in the last call of this.appendComment
+
+  //This adjusts this.insertPoint according to timestamp value
+  moveInsertPoint: function(timestamp, commentBlocks) {
+    if(commentBlocks.length === 0) {
+      return;
```

Comment:
You will have to set `this.insertPoint = 0` first. Otherwise when you delete the only comment and try to comment afterwards the comment won't be added to the comment stream. Please also add a test for that case.